src/hwc/compile/luna/luna.go

```diff
@@ -0,0 +1,238 @@
+			if err != nil {
+				continue
+			}
+			var credentials LunaCredentials
+			err = json.Unmarshal(credentialsJson, &credentials)
+			if err == nil && validateCredentials(&credentials) {
+				l.credentials = &credentials
+				return
+			}
+		}
+	}
+}
+
+func (l *Luna) createLunaDirectory() error {
+	return os.MkdirAll(l.LunaDir, 0755)
+}
+
+func (l *Luna) writeNTLSCredentials() error {
+
+	privateKey := []byte(l.credentials.Client.PrivateKey)
+	err := ioutil.WriteFile(l.ClientPrivateKeyPath, privateKey, 0640)
+	if err != nil {
+		return err
+	}
+
+	clientCertificate := []byte(l.credentials.Client.Certificate)
+	err = ioutil.WriteFile(l.ClientCertificatePath, clientCertificate, 0644)
+	if err != nil {
+		return err
+	}
+
+	var serverCertificates string
+	serverCertificates = ""
+	for _, server := range l.credentials.Servers {
+		serverCertificates += server.Certificate
+	}
+	err = ioutil.WriteFile(l.ServerCertificatesPath, []byte(serverCertificates), 0644)
+
+	return err
+}
+
+func (l *Luna) writeCrystokiIni() error {
+	var conf string
+	conf = ""
+	lunaSection := `[Luna]
+PEDTimeout1=100000
+PEDTimeout2=200000
+CommandTimeoutPedSet=720000
+KeypairGenTimeOut=2700000
+CloningCommandTimeOut=300000
+PEDTimeout3=10000
+DefaultTimeOut=500000
+`
+
+	conf += lunaSection
+
+	client := `[LunaSA Client]
+ReceiveTimeout=20000
+TCPKeepAlive=1
+NetClient=1
+`
+	client += "ServerCAFile=.cloudfoundry\\.luna\\serverCertificates.pem\n"
+	client += "ClientCertFile=.cloudfoundry\\.luna\\clientCertificate.pem\n"
+	client += "ClientPrivKeyFile=.cloudfoundry\\.luna\\clientPrivateKey.pem\n"
+
+	for index, server := range l.credentials.Servers {
+		client += fmt.Sprintf("ServerName%02d=", index) + server.Name + "\n"
+		client += fmt.Sprintf("ServerPort%02d=1792", index) + "\n"
+		client += fmt.Sprintf("ServerHtl%02d=0", index) + "\n"
+	}
+
+	conf += client
+
+	misc := `[Misc]
+PE1746Enabled=0
+`
+
+	conf += misc
+
+	conf += "[HASynchronize]\n"
+	for _, group := range l.credentials.Groups {
+		conf += group.Label + "=1\n"
+	}
+
+	haconf := `[HAConfiguration]
+HAOnly=1
+reconnAtt=-1
+AutoReconnectInterval=60
+`
+	conf += haconf
+	conf += "[VirtualToken]\n"
+	for index, group := range l.credentials.Groups {
+		conf += fmt.Sprintf("VirtualToken%02dLabel=%s\n", index, group.Label)
+		conf += fmt.Sprintf("VirtualToken%02dSN=1%s\n", index, group.Members[0])
+		members := strings.Join(group.Members, ",")
+		conf += fmt.Sprintf("VirtualToken%02dMembers=%s\n", index, members)
+	}
+
+	filepath.Join(l.LunaDir, "crystoki.ini")
+	conf = strings.Replace(conf, "\n", "\r\n", -1)
+	err := ioutil.WriteFile(filepath.Join(l.LunaDir, "crystoki.ini"), []byte(conf), 0666)
+	if err != nil {
+		return err
+	}
+	return err
+}
+
+func (l *Luna) writeProfileD() error {
+	profileD := filepath.Join(l.BuildDir, ".profile.d")
+	err := os.MkdirAll(profileD, 0755)
+	if err != nil {
+		return err
+	}
+	lunaBatch := "set ChrystokiConfigurationPath=.cloudfoundry\\.luna\r\n"
```

Comment:
This spelling, `ChrystokiConfigurationPath` (starting with `Chr`) seems inconsistent with the references to `crystoki.ini` or `cryptoki.dll`. Is `Chrystoki` supposed to be different from `crystoki`?

Reply:
This spelling difference is intentional, and I agree, not consistent.  The ```ini``` file should have been named ```chrystoki.ini``` (on linux it is ```Chrystoki.conf```) and the ```Chrys``` prefix comes from "Chrysalis ITS".  This would make it consistent but it has stayed the same to remain backwards compatible and the name is not configurable.